spec/capitalize_decorator_spec.rb

```diff
@@ -1,13 +1,48 @@
 require_relative '../class/capitalize_decorator'
+require_relative '../class/decorator'
+require_relative '../class/nameable'
 
 describe CapitalizeDecorator do
   before :each do
-    @dummy_capitalize_decorator = CapitalizeDecorator.new('Fickry')
+    @dummy_capitalize_decorator = CapitalizeDecorator.new('john doe')
   end
 
   describe '-> Initialize class (.new)' do
     it 'should be create an instance of CapitalizeDecorator class' do
       expect(@dummy_capitalize_decorator).to be_an_instance_of CapitalizeDecorator
     end
   end
+
+  describe '-> Method (.correct_name)' do
+    it 'should change the first letter of name is uppercase' do
+      nameable = double('nameable')
+      decorated_nameable = CapitalizeDecorator.new(nameable)
+
+      allow(nameable).to receive(:correct_name).and_return('fickry')
+      result = decorated_nameable.correct_name
+      expect(result).to eql('Fickry')
+    end
+
+    it 'returns the correct name from the decorated object without truncation' do
+      nameable = double('nameable')
+      decorated_nameable = CapitalizeDecorator.new(nameable)
+
+      allow(nameable).to receive(:correct_name).and_return('john')
+
+      result = decorated_nameable.correct_name
+
+      expect(result).to eq('John')
+    end
+
+    it 'returns the correct name from the decorated object with truncation' do
+      nameable = double('nameable')
+      decorated_nameable = CapitalizeDecorator.new(nameable)
+
+      allow(nameable).to receive(:correct_name).and_return('Lorem ipsum dolor')
+
+      result = decorated_nameable.correct_name
+
+      expect(result).to eq('Lorem ipsu')
+    end
```

Comment:
- [ ] I think there's some confusion here as trimming the given name to only 10 characters is the responsibility of the `trimmer_decorator` class and should be tested in the `trimmer_decorator_spec` accordingly as mentioned previously. So as such, I'd suggest adding this test to the proper file and removing the implementation of trimming strings from the `capitalize_decorator` class as it's only responsible for capitalizing the first character of a given string.

Reply:
ok, moved the trimming for the subtract name to only 10 characters into the trimmer_decorator class